File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        
        int count = 0;
        
        for(int i=0; i<babbling.length; i++){
            
            if(babbling[i].contains("ayaaya") || babbling[i].contains("yeye")||babbling[i].contains("mama")||babbling[i].contains("woowoo")){
                continue;
            }
            babbling[i] = babbling[i].replace("aya", "true");
            babbling[i] = babbling[i].replace("ye", "true");
            babbling[i] = babbling[i].replace("ma", "true");
            babbling[i] = babbling[i].replace("woo", "true");
            babbling[i] = babbling[i].replace("true", "");
            
            
            if(babbling[i].length() == 0){
                count += 1;
            }
        }
        
        
        return count;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    
    let count = 0;
    
    for(let str of babbling){
        
        if(str.includes('ayaaya') || str.includes('yeye') || str.includes('woowoo') || str.includes('mama')){
            continue;
        }
        
        str = str.replaceAll('aya', 'true');
        str = str.replaceAll('ye', 'true');
        str = str.replaceAll('ma', 'true');
        str = str.replaceAll('woo', 'true');
        str = str.replaceAll('true', '');
                
        if(str.length == 0){
            count++;
        }
    }
    
    return count;
}